### PR TITLE
Add meson.build

### PR DESCRIPTION
**Pull request by DJATOM**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,35 @@
+project('L-SMASH-Works', 'cpp',
```
Comment (myrsloik) [inline meson.build:1]: Excuse me?

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  default_options : ['buildtype=release', 'b_ndebug=if-release', 'c_std=c++17'],
```
Comment (myrsloik) [inline meson.build:2]: Did you even look at the required C++ version?

File: meson.build [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  meson_version : '>=0.48.0'
+)
+
+add_project_arguments('-ffast-math', language : 'cpp')
```
Comment (myrsloik) [inline meson.build:6]: Nope

File: meson.build (hunk 1 above)
Comment (DJATOM) [reply meson.build:1]: It's edited from other plugin meson file. So I just overlook that, will fix.

File: meson.build (hunk 2 above)
Comment (DJATOM) [reply meson.build:2]: [Yes?](https://github.com/vapoursynth/bestaudiosource/blob/master/msvc/BestAudioSource.vcxproj#L166)

File: meson.build (hunk 3 above)
Comment (DJATOM) [reply meson.build]: Removed from the file

File: meson.build [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+vapoursynth_dep = dependency('vapoursynth').partial_dependency(compile_args : true, includes : true)
+
+deps = [
+  vapoursynth_dep,
+  dependency('libavcodec', version : '>=58.18.0'),
+  dependency('libavformat', version : '>=58.12.0')
+]
+
+if host_machine.cpu_family().startswith('x86')
```
Comment (myrsloik) [inline meson.build:20]: Pointless and should go

File: meson.build [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  add_project_arguments('-mfpmath=sse', '-msse2', language : 'cpp')
+endif
+
+if host_machine.system() == 'windows'
```
Comment (myrsloik) [inline meson.build:24]: Should also go since I don't support mingw for anything ever

File: meson.build (hunk 4 above)
Comment (DJATOM) [reply meson.build]: Done.

File: meson.build (hunk 5 above)
Comment (DJATOM) [reply meson.build]: Also removed.

**Pull request merged**